### ISSUE TITLE
Parameter envelope is not taken in consideration in @Validation on requestHandler accepting SQSEvent

### ISSUE THREAD

**Issue by chihebdaoues** [awslabs/aws-lambda-powertools-java#959]
My team is trying to use `aws-lambda-powertools' validation utility to validate request/response schemas on a couple of lambda functions we have and we noticed that the envelope parameter is not being taken in consideration when provided.

**What were you trying to accomplish?**
We have an SQS queue that subscribes to an SNS topic and a lambda function that subscribes to the SQS queue. Which I think is fairly common pattern. I was trying to annotate the request handler with validate to put some validation on the event that originated from the SNS topic schema and as such I needed to use the envelope parameter.

```
@Validation(inboundSchema = "classpath:/schemas/requestSchema.json", envelope = "records[*].powertools_json(body).powertools_json(Message)")
    public Void requestHandler(SQSEvent event, Context context) {
....
```

## Expected Behavior
I expected the Validation to use the envelope provided, but I think powertools is actually using the envelope for SQSEvents. I was expecting the default behavior to be overridden by the provided envelope.

## Current Behavior
Looking at the code in [ValidationAspect.java#L108](https://github.com/awslabs/aws-lambda-powertools-java/blob/bcf9ababdfe64fda62ff023ee36a18825f976165/powertools-validation/src/main/java/software/amazon/lambda/powertools/validation/internal/ValidationAspect.java#L108), I see that the provided envelope will be used as the last resort. And since my event is an SQSEvent, it evaluates to true on [ValidationAspect.java#L71](https://github.com/awslabs/aws-lambda-powertools-java/blob/bcf9ababdfe64fda62ff023ee36a18825f976165/powertools-validation/src/main/java/software/amazon/lambda/powertools/validation/internal/ValidationAspect.java#L71) and as such my envelope is not being used.

The current behavior is enforcing a one solution fits all for validating all the [builtin events](https://awslabs.github.io/aws-lambda-powertools-java/utilities/validation/#built-in-events-and-responses).

## Possible Solution
Test if the envelope is not empty at the start of the `if...else` block and if so use it. This would allow the user to override the default behavior which enables clients to customize the validation as they want.

## Steps to Reproduce (for bugs)
1. Create an SNS topic
2. Subscribe to the topic using an SQS queue
3. Create a lambda that subscribes to the SQS queue
4. Try to validate the schema of message that originated from the SNS topic

## Environment

* **Powertools version used**: 1.12.2
* **Packaging format (Layers, Maven/Gradle)**: Gradle
* **AWS Lambda function runtime:** JAVA11

**Comment by jeromevdl** [comment]
Hi @chihebdaoues,

Thanks for noticing this. I agree, if the envelope is set, it should take precedence over the defaults.

**Comment by msailes** [comment]
This is available in v1.13.0
